### DISCUSSION TITLE
Choose Folder and create file by template

### DISCUSSION THREAD

**Discussion by yleen** [SilentVoid13/Templater#698] [Templates Showcase]
```javascript
<%*
const folders = this.app.vault.getAllLoadedFiles().filter(i => i.children).map(folder => folder.path);
const folderPath = await tp.system.suggester(folders, folders);
const title = await tp.system.prompt("Title")
await tp.file.rename(`${title}`)
await tp.file.move(`/${folderPath}/${title}`)
%>
```

**Comment by dredhorse** [comment]
you can neglect the tp.file.rename... the move will do a rename too.

**Comment by floris-max** [comment]
<%*
try {
    // Obtener todas las carpetas del sistema
    const folders = this.app.vault.getAllLoadedFiles()
        .filter(item => item.children) // Filtrar solo carpetas
        .map(folder => folder.path); // Obtener las rutas de las carpetas

    // Validar que haya carpetas disponibles
    if (folders.length === 0) {
        throw new Error("No se encontraron carpetas en el sistema.");
    }

    // Mostrar sugeridor para seleccionar una carpeta
    const folderPath = await tp.system.suggester(folders, folders);
    if (!folderPath) {
        throw new Error("Debe seleccionar una carpeta.");
    }

    // Solicitar al usuario un título para el archivo
    const title = await tp.system.prompt("Título de la nota:");
    if (!title || title.trim() === "") {
        throw new Error("El título no puede estar vacío.");
    }

    // Generar el nuevo nombre y la ruta del archivo
    const fileName = `${title.trim()}`;
    const destinationPath = `/${folderPath}/${fileName}`;

    // Verificar si ya existe un archivo con el mismo nombre en la carpeta seleccionada
    if (await this.app.vault.adapter.exists(destinationPath)) {
        throw new Error(`Ya existe un archivo llamado "${fileName}" en la carpeta seleccionada.`);
    }

    // Renombrar el archivo actual y moverlo a la carpeta seleccionada
    await tp.file.rename(fileName);
    await tp.file.move(destinationPath);

    // Confirmación de éxito
    console.log(`Archivo movido exitosamente a: ${destinationPath}`);
} catch (error) {
    // Mostrar mensajes de error en la consola
    console.error("Error al mover el archivo:", error.message);
    new Notice(`Error: ${error.message}`);
}
%>
